### DISCUSSION TITLE
berry script blocks device?

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#18022] [Q&A]
i was testing with for loops.

on last try i forgot a the break that shorten the for-loop.
after rebooting the device never came back again (WebUI).
The device is pingable over wifi.
Serial Port is there. Output:

![grafik](https://user-images.githubusercontent.com/16272493/220637490-b03f1300-78b8-48e5-9f73-22bdded8cfd4.png)

can i acces the filesystem by serial port?
or is there an option to deactivate berry script until code was fixed?

I think it does to much blocking delay...

```
def read_sensor_average()
       var temp_value
       var temp_string=" "
       var temp_value_average

     # read_sensor()     #fill GLOBAL_vars
       GLOBAL_value=3.48          #set fix value since there is no sensor
       GLOBAL_value_floating_average = 3.68

        temp_value=10*GLOBAL_value
        temp_value_average=10*GLOBAL_value_floating_average

        for i: 0 .. 150
            if i<temp_value
                temp_string= temp_string + " " 
            else i==temp_value
                temp_string= temp_string + "|" 
                 
                for j: 0 .. 150
                    if j < (temp_value_average - temp_string  +20)
                        temp_string= temp_string + " "  
                    else
                         temp_string= temp_string + "|" 
                         #break    # i forgot to uncomment this break
                    end
                end     
                
            end 
            
        end

        print(temp_string)

    tasmota.set_timer(1000, read_sensor_average)  #do every sec

end

read_sensor_average()

```

**Comment by sfromis** [comment]
You should just not have "slow" loops in Berry code, always exit quickly, as the code is called during the Tasmota main loop, and Tasmota is blocked until you exit.

**Reply by sfromis** [reply]
Easiest recovery is wired erase/flash, I think that it would be "less realistic" to try to get a command to delete/rename in before `autoexec.be`  starts. Of course, this would lose your file system. If you really, *really*  want to recover files, you could build a binary not including Berry to have it boot without looping again, and flash that binary *without erase*.
